modules/core/src/main/java/org/apache/ignite/internal/processors/cache/warmup/LoadAllWarmUp.java

```diff
@@ -0,0 +1,207 @@
+
+/**
+ * "Load all" warm-up strategy, which loads pages to persistent data region
+ * until it reaches {@link DataRegionConfiguration#getMaxSize} with priority
+ * to index partitions. Loading occurs sequentially for each cache group.
+ */
+public class LoadAllWarmUp implements WarmUpStrategy<LoadAllWarmUpConfiguration> {
+    /** Logger. */
+    private final IgniteLogger log;
+
+    /**
+     * Cache group contexts supplier.
+     * Since {@link GridCacheProcessor} starts later.
+     */
+    private final Supplier<Collection<CacheGroupContext>> grpCtxSup;
+
+    /** Stop flag. */
+    private volatile boolean stop;
+
+    /**
+     * Constructor.
+     *
+     * @param log Logger.
+     * @param grpCtxSup Cache group contexts supplier. Since {@link GridCacheProcessor} starts later.
+     */
+    public LoadAllWarmUp(IgniteLogger log, Supplier<Collection<CacheGroupContext>> grpCtxSup) {
+        this.log = log;
+        this.grpCtxSup = grpCtxSup;
+    }
+
+    /** {@inheritDoc} */
+    @Override public Class<LoadAllWarmUpConfiguration> configClass() {
+        return LoadAllWarmUpConfiguration.class;
+    }
+
+    /** {@inheritDoc} */
+    @Override public void warmUp(
+        LoadAllWarmUpConfiguration cfg,
+        DataRegion region
+    ) throws IgniteCheckedException {
+        if (stop)
+            return;
+
+        assert region.config().isPersistenceEnabled();
+
+        Map<CacheGroupContext, T2<Integer, Long>> loadDataInfo = loadDataInfo(region);
+
+        if (log.isInfoEnabled()) {
+            log.info("Order of cache groups loaded into data region [name=" + region.config().getName()
+                + ", partCnt=" + loadDataInfo.values().stream().mapToLong(T2::get1).sum()
+                + ", pageCnt=" + loadDataInfo.values().stream().mapToLong(T2::get2).sum()
+                + ", availablePageCnt=" + availableLoadPageCount(region) + ", grpNames=" +
+                loadDataInfo.keySet().stream().map(CacheGroupContext::cacheOrGroupName).collect(toList()) + ']');
+        }
+
+        long allPartCnt = 0;
+        long allPageCnt = 0;
+
+        for (Map.Entry<CacheGroupContext, T2<Integer, Long>> e : loadDataInfo.entrySet()) {
+            CacheGroupContext grp = e.getKey();
+
+            // Including index partition and its pages.
+            final int partCnt = e.getValue().get1();
+            final long pageCnt = e.getValue().get2();
+
+            if (log.isInfoEnabled()) {
+                log.info("Start warm-up cache group, starting from index [name=" + grp.cacheOrGroupName()
+                    + ", partCnt=" + partCnt + ", pageCnt=" + pageCnt + ']');
+            }
+
+            PageMemoryEx pageMemEx = (PageMemoryEx)region.pageMemory();
+
+            long grpPageCnt = 0;
+
+            // Starting with index.
+            for (int i = -1; i < partCnt - 1; i++, allPartCnt++) {
+                long pageId = pageMemEx.partitionMetaPageId(grp.groupId(), (i == -1 ? INDEX_PARTITION : i));
```

Comment:
I didn't get how do you handle gaps in partitions? e.g. you have local partitions 1,3,5 - it is 3 in total but you can't iterate from 1 to 3